src/bucket/BucketOutputIterator.cpp

```diff
@@ -79,10 +80,9 @@ BucketOutputIterator::put(BucketEntry const& e)
 std::shared_ptr<Bucket>
 BucketOutputIterator::getBucket(BucketManager& bucketManager)
 {
-    assert(mOut);
```

Comment:
Btw, I removed this assert, because I don't think we can actually guarantee this. The stream might not be flushed causing the call to `writeOne` succeed, while the actual disk write later fails. We should still be able to detect problems, since `writeAndThrowOnFail` and `close` will throw.

Reply:
`close` does not throw by default